src/matrix/mod.rs

```diff
@@ -278,7 +266,7 @@ impl JsonPostError {
 
 quick_error! {
     #[derive(Debug)]
-    pub enum LoginError {
+    pub enum Error {
```

Comment:
In order to remove the `.unwrap()`s from `join_room()` I unified all the error possibilities under one type. Previously this was done by many `.map_err()`s, and I thought this made things more readable. Thoughts on this?

Reply:
I think that's fine